lookout/style/typos/corrector.py

```diff
@@ -77,118 +82,110 @@ def initialize_generator(self, vocabulary_file: str, frequencies_file: str,
-                typos, self.threads_number, save_candidates_file)
-        self.ranker.fit(typos[Columns.CorrectToken], get_candidates_metadata(candidates),
+                data, self.threads_number, save_candidates_file)
+        else:
+            candidates = pandas.read_csv(candidates, index_col=0)
+        self.ranker.fit(data[Columns.CorrectToken], get_candidates_metadata(candidates),
                         get_candidates_features(candidates))
 
-    def train_on_file(self, typos_file: str, candidates_file: str = None,
+    def train_on_file(self, data_file: str, candidates:  str = None,
                       save_candidates_file: str = None) -> None:
         """
-        Train corrector on the given dataset of typos inside identifiers.
+        Train corrector on tokens from the given file.
 
-        :param typos_file: CSV file with columns "typo" and "identifier",
-                           column "token_split" is optional, but used when present.
-        :param candidates_file: Pickle dump of pandas.DataFrame with precalculated \
-                                candidates and features
-        :param save_candidates_file: Path to file where to save the candidates.
+        :param data_file: A .csv dump of a dataframe, containing columns Columns.Token and \
+                          Columns.CorrectToken, column Columns.Split is optional, \
+                          but used when present.
+        :param candidates: A .csv.xz dump of a dataframe with precalculated candidates.
+        :param save_candidates_file: Path to file where to save the candidates (.csv.xz).
         """
-        typos = pandas.read_csv(typos_file, index_col=0)
-        candidates = None
-        if candidates_file is not None:
-            candidates = pandas.read_pickle(candidates_file)
-        self.train(typos, candidates, save_candidates_file)
+        self.train(pandas.read_csv(data_file, index_col=0), candidates, save_candidates_file)
 
-    def suggest(self, typos: pandas.DataFrame, candidates: pandas.DataFrame = None,
+    def suggest(self, data: pandas.DataFrame, candidates: str = None,
                 save_candidates_file: str = None, n_candidates: int = 3,
                 return_all: bool = True) -> Dict[int, List[Tuple[str, float]]]:
         """
-        Suggest corrections for given typos.
+        Suggest corrections for the tokens from the given dataset.
 
-        :param typos: DataFrame containing column "typo", \
-                      column "token_split" is optional, but used when present
-        :param candidates: DataFrame with precalculated candidates
-        :param n_candidates: Number of most probable candidates to return
-        :param return_all: False to return suggestions only for corrected tokens
-        :param save_candidates_file: Path to file to save candidates to
+        :param data: DataFrame, containing column Columns.Token, column Columns.Split \
+                     is optional, but used when present.
```

Comment:
I really don't like this optionality. Do we use it in our normal analysis flow? If we do, make it required. If we don't, remove it.